src/enumerable.rb

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    length.times { |i| yield(self[i]) }
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    length.times { |i| yield(self[i], i) }
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    selected = []
+    my_each { |obj| selected << obj if yield(obj) }
+    selected
+  end
+
+  def my_all?(pattern = nil)
+    if block_given?
+      my_each { |obj| return false unless yield(obj) }
+    elsif pattern
+      my_each do |obj|
+        return false unless pattern_match?(obj, pattern)
+      end
+    else
+      my_each { |obj| return false unless obj }
+      true
+    end
+    true
+  end
+
+  def my_any?(pattern = nil)
+    if block_given?
+      my_each { |obj| return true if yield(obj) }
+    elsif pattern
+      my_each do |obj|
+        return true if pattern_match?(obj, pattern)
+      end
+    else
+      my_each { |obj| return true if obj }
+    end
+    false
+  end
+
+  def my_none?(pattern = nil)
+    if block_given?
+      my_each { |obj| return false if yield(obj) }
+    elsif pattern
```

Comment:
**[REQUIRED]**
When a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern.
`[1,2,3,4].my_none?(5) => returns true`
`[1,2,3,4,5].my_none?(5) => returns false`